drivers/gpu/drm/vc4/vc4_hdmi.c

```diff
@@ -991,7 +991,11 @@ static int vc4_hdmi_audio_prepare(struct snd_pcm_substream *substream,
 	vc4_hdmi_audio_set_mai_clock(vc4_hdmi);
 
 	mai_sample_rate = sample_rate_to_mai_fmt(vc4_hdmi->audio.samplerate);
-	mai_audio_format = 2; /* PCM mode */
+	if (vc4_hdmi->audio.iec_status[0] & IEC958_AES0_NONAUDIO &&
+		vc4_hdmi->audio.channels == 8)
+		mai_audio_format = 200; /* HBR mode */
```

Comment:
proper defines in vc4_regs.h instead of magic number here and below would be nice